packages/web-api/src/methods.ts

```diff
@@ -1985,263 +2048,160 @@ interface FileUploadComplete {
-}
-
-/*
- * `im.*`
- */
-export interface IMCloseArguments extends WebAPICallOptions, TokenOverridable {
-  channel: string;
-}
-export interface IMHistoryArguments extends WebAPICallOptions, TokenOverridable, TimelinePaginationEnabled {
-  channel: string;
-  count?: number;
-  unreads?: boolean;
-}
-export interface IMListArguments extends WebAPICallOptions, TokenOverridable, CursorPaginationEnabled { }
-cursorPaginationEnabledMethods.add('im.list');
-export interface IMMarkArguments extends WebAPICallOptions, TokenOverridable {
-  channel: string;
-  ts: string;
-}
-export interface IMOpenArguments extends WebAPICallOptions, TokenOverridable, LocaleAware {
-  user: string;
-  return_im?: boolean;
-}
-export interface IMRepliesArguments extends WebAPICallOptions, TokenOverridable {
-  channel: string;
-  thread_ts?: string;
-}
-
 /*
  * `migration.*`
  */
-export interface MigrationExchangeArguments extends WebAPICallOptions, TokenOverridable {
+// https://api.slack.com/methods/migration.exchange
+export interface MigrationExchangeArguments extends TokenOverridable, OptionalTeamAssignable {
   users: string; // comma-separated list of users
   to_old?: boolean;
-  team_id?: string;
-}
-
-/*
- * `mpim.*`
- */
-export interface MPIMCloseArguments extends WebAPICallOptions, TokenOverridable {
-  channel: string;
-}
-export interface MPIMHistoryArguments extends WebAPICallOptions, TokenOverridable, TimelinePaginationEnabled {
-  channel: string;
-  count?: number;
-  unreads?: boolean;
-}
-export interface MPIMListArguments extends WebAPICallOptions, TokenOverridable, CursorPaginationEnabled { }
-cursorPaginationEnabledMethods.add('mpim.list');
-export interface MPIMMarkArguments extends WebAPICallOptions, TokenOverridable {
-  channel: string;
-  ts: string;
-}
-export interface MPIMOpenArguments extends WebAPICallOptions, TokenOverridable {
-  users: string; // comma-separated list of users
-}
-export interface MPIMRepliesArguments extends WebAPICallOptions, TokenOverridable {
-  channel: string;
-  thread_ts: string;
 }
 
 /*
  * `oauth.*`
  */
-export interface OAuthAccessArguments extends WebAPICallOptions {
-  client_id: string;
-  client_secret: string;
-  code: string;
+// https://api.slack.com/methods/oauth.access
+// TODO: this method is marked as a 'legacy' method; should we add it as a 'deprecated' method?
+export interface OAuthAccessArguments {
+  client_id: string; // TODO: docs state this is optional
+  client_secret: string; // TODO: docs state this is optional
+  code: string; // TODO: docs state this is optional
   redirect_uri?: string;
-  single_channel?: string;
+  single_channel?: string; // TODO: docs state this is a boolean, only valid for legacy apps
 }
-export interface OAuthV2AccessArguments extends WebAPICallOptions {
-  client_id: string;
-  client_secret: string;
+// https://api.slack.com/methods/oauth.v2.access
+export interface OAuthV2AccessArguments {
+  client_id: string; // TODO: docs state this is optional
```

Comment:
The document might be wrong; I don't know what the possible scenario to call this API without the pair could be